driver-core/src/main/java/com/datastax/driver/core/DataTypeCqlNameParser.java

```diff
@@ -148,7 +148,7 @@ static DataType parse(String toParse, Cluster cluster, String currentKeyspaceNam
         if (userType == null)
             throw new UnresolvedUserTypeException(currentKeyspaceName, type);
         else
-            return userType;
+            return userType.copy(frozen);
```

Comment:
Just curious, but if the UDT is already resolved/known, isn't it likely that the `frozen` field on that existing user type is already correct?